netsnmp/__init__.py

```diff
@@ -60,14 +60,13 @@ def oid_outside_tree(oid1, oid2):
     Return True if oid2 outside of oid1 tree
     Used by SNMPSession.walk
     """
-    oid1_split = [i for i in oid1.split('.') if i]
-    oid1_idx = len(oid1_split)-1
-
-    oid2_split = [i for i in oid2.split('.') if i]
-
-    if oid2_split[oid1_idx] > oid1_split[oid1_idx]:
+    try:
+        if oid2.startswith(oid1):
+            return False
+        return True
+    except:
         return True
-    return False
+
```

Comment:
This won't work as expected.

for example:

```
>>> oid1 = "1.3.6.1.2.1.1.1.0"
>>> oid2 = "1.3.6.1.2.1.1.1.1"
>>> oid2.startswith(oid1)
False
```

This should return True as those are within the same "tree". We could do `startswith()` but would still have to split/re-join oid1 - we might as well just leave this code untouched. It is a near 1:1 copy of the net-snmp methodology.